### ISSUE TITLE
Refactor the Boutiques integrator

### ISSUE THREAD

**Issue by prioux** [aces/cbrain#1130]
(Special preamble note: I was sure there was already issue for that, because I've been thinking about this for over a year, yet I can't find one)

The current Boutiques integration works with templates OF templates (`e.g. portal.html.erb.erb`). 

A better integration would be to create a base cbrain task class CbrainTask::Boutiques integration that would have a single set of normal templates, but these templates would execute their control code by directly accessing the Boutiques descriptor at run-time. A descriptor for tool Xyz would generate dynamically a very simple class like this:

```ruby
class CbrainTask::Xyz < CbrainTask::Boutiques
   boutique_descriptor_path 'xyz.json'
end
```
(Not literally evaled(), using meta programming of course) 

We'd need to code a new Ruby class to represent a Boutiques descriptor, and a set of web page generators that take subsets of that descriptor to produce page sections. The portal code would contain something like

```erb
<% descriptor.inputs.each do |input| %>
  <%= generate_input(input) %>
<% end %>
```

Many front-end elements (html and javascript) already implemented in the current templated templates could be reused and simplified.

The very nice thing about this is that new Boutiques tasks with special requirements could inherit from a different class than "CbrainTask::Boutiques, as specified in the descriptor's `custom` section.

**Comment by natacha-beck** [comment]
@prioux this is done by your new Boutiques integrator?

**Comment by prioux** [comment]
Hi @natacha-beck you are absolutely right, this is the issue I created about the new framework.

Implemented in #1153
